Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java

```diff
@@ -206,16 +206,100 @@ public void onResponse(Call call, final Response response) throws IOException {
+                .build();
+        //Setup callback
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                Log.e("Request Failed", "Message : " + e.getMessage());
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) {
+                mHandler.post(() -> {
+                    if (response.isSuccessful()) {
+                        try {
+                            String res = response.body().string();
+                            Log.v("RESULT", res);
+                            JSONArray ar = new JSONArray(res);
+                            for (int i = 0; i < ar.length(); i++) {
+                                mSearchCities.add(new HotelSearchModel(
+                                        ar.getJSONObject(i).getString("city_name"),
+                                        ar.getJSONObject(i).optString("image"),
+                                        ar.getJSONObject(i).getString("id")));
+                            }
+                        } catch (JSONException | IOException e) {
+                            e.printStackTrace();
+                            Log.e("ERROR", "Message : " + e.getMessage());
+                        }
+                    } else {
+                        Log.e("ERROR", "Network error");
+                    }
+                    pb.setVisibility(View.GONE);
+                });
+            }
+        });
+
+    }
+    /**
+     * Calls the API & fetch details of city with given id
+     *
+     * @param cityId the city id
+     */
+    public void getCityInfo(String cityId) {
+
+        pb.setVisibility(View.VISIBLE);
+
+        String uri = API_LINK_V2 + "get-city/" + cityId;
+        OkHttpClient client = new OkHttpClient();
+
+        final Request request = new Request.Builder()
+                .header("Authorization", "Token " + mToken)
+                .url(uri)
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+
+            }
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                final String res = Objects.requireNonNull(response.body()).string();
+                try {
+                    Log.v("Response", res);
+                    JSONObject responseObject = new JSONObject(res);
+                    mDeslat = responseObject.getString("latitude");
+                    mDeslon = responseObject.getString("longitude");
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+        getHotelList();
+    }
+
     @Override
     protected void onResume() {
         super.onResume();
         mSource = mSharedPreferences.getString(SOURCE_CITY_ID, "1");
```

Comment:
you can remove these 4 variables now. I don't think these are being used anywhere now:
```
private String mSource;
    private String mDestination;
    private String mSourceText;
    private String mDestinationText;
```